### DISCUSSION TITLE
Adding custom <style> in the head tag

### DISCUSSION THREAD

**Discussion by IonelLupu** [tinymce/tinymce#7806] [General]
Hi,

We are writing/editing some email templates using tinyMce and GMail wants the CSS styles to be in a <style> tag inside the <head> tag of the email template in order to not strip it from the email. We can't use inline styles because we have to write media queries. In tinyMce 6 there is no fullpage plugin anymore and we can't see the `<html>`, `<head>` and `<body>` tags. How can we make tinyMce display these tags so we can add our `<style>` tag in the `<head>`?

**Comment by lostkeys** [comment]
Hi @IonelLupu. I believe the best approach is to post process your HTML outside of TinyMCE and use some tool to generate email friendly output in the format you require.

**Reply by IonelLupu** [reply]
I assume TinyMCE doesn't allow editing the full HTML anymore (the `<head>` tag I mean) because there is no "fullpage" plugin since v6, right? We would like not to do postprocessing this way as it will get messy really fast.

What do you mean to use some tool to generate friendly email output? Do you mean replacing TinyMCE altogether and use another editor for editing email templates? 

TinyMCE looks really capable as an editor. We just want to be able to write the `<style>` tag in the `<head>`. In v5, the plugin `fullpage` allowed us to do that but now, in v6, it was removed. I saw it was recommended to fork the  'fullpage' plugin but I was wondering if there is a solution provided by TinyMCE v6 instead.

Btw, why can't we edit the entire HTML document (with `<html>`, `<head>` and `<body>`)? Can a setting be added so it shows these tags?

**Reply by lostkeys** [reply]
The full page plugin was removed because it caused a lot of bugs and issues. I don't have all the details in front of me but we wouldn't have removed it if it was working alright.

TinyMCE is a very capable WYSIWYG HTML editor, but it is for the HTML you have between the body tags.

What I mean with post process is to find (or write) a tool that take the content from TinyMCE and generate a complete HTML document suitable for emailing.

**Reply by IonelLupu** [reply]
In this case, we will do just that: on the backend side, before sending the email, we will take all the `<style>`  tags and move them to the `<head>` tag manually. Thank you for your answer! 😄
